### PR TITLE
Fixed #571 Added Service Worker (PWA Feature).

### LINKED ISSUE #571: Adding PWA Features to the Site
### Expected results
Should Load Faster
### Actual results
Loads Very Slowly
### Steps to reproduce the issue
Add Service Workers and Cashing
## Would you like to work on this
Yes.

### PR DESCRIPTION
**Pull request by niranjannitesh**
Please read and understand everything below
**Do not delete any text other than where you are instructed.**

**Students: If one of them is applicable to you. Please check it.**

Check by changing each `[ ]` to `[x]` Please take note of the whitespace as it matters.

- [ ] Read and understood (see CONTRIBUTING.md)
- [x] Included a Preview link and screenshot showing after and before the changes.
- [ ] Images are `240 x 240` [w x h].
- [x] Included a description of change below.
- [ ] Squashed the commits.

# Changes done in this Pull Request
Added Service Worker to the website to improve site speed.
Preview URL: https://niteshkumarniranjan.github.io/gci17.fossasia.org/

- Fixes `#571`

### CODE REVIEW

**Comment by niranjannitesh** [comment]
For Images Visit #721

**Comment by harshit98** [comment]
@niteshkumarniranjan PR title should be meaningful and should be related to the fix done.

**Comment by niranjannitesh** [comment]
@harshit98 Now!

**Review by vaibhavsingh97 — APPROVED** [review]

**Review by samyakbvs — CHANGES_REQUESTED** [review]
Pls squash your commits

**Review by vaibhavsingh97 — COMMENTED** [review]
Please squash your commits and submit test result of before and after PWA using [this](https://tools.pingdom.com/) tool

**Comment by niranjannitesh** [comment]
Test Results are at https://github.com/fossasia/gci17.fossasia.org/pull/721

**Comment by niranjannitesh** [comment]
@vaibhavsingh97 Squashed!

**Review by dilpreetsio — COMMENTED** [review]
@niteshkumarniranjan Add a manifest file Please have a look at the screenshot

screen shot 2018-01-02 at 12 21 39 am

https://developers.google.com/web/progressive-web-apps/checklist

**Comment by niranjannitesh** [comment]
@geekyd I had tried to implement the service worker so that the site works fast. If I will be Implementing all the features then I have to write the whole site from start.

**Comment by niranjannitesh** [comment]
@harshit98

**Comment by dilpreetsio** [comment]
@niteshkumarniranjan No you do not have to re-do the entire site. You need to add a manifest.json file. Have a look at this https://github.com/fossasia/open-event-webapp/commit/ab6c5ec21863bdb020d9d256b2eb857601f84a9b#diff-b156726f07a174b5784fad30dd330df6. We need to add something similar to this here also.

Hope this helps.

**Comment by niranjannitesh** [comment]
Okay! I am adding it.

File: manifest.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+    "start_url": "index.html",
+    "scope": ".",
+    "display": "standalone",
+    "orientation": "portrait-primary",
+    "background_color": "#fff",
+    "theme_color": "#3f51b5",
+    "description": "Open Event Web Application Generator",
+    "dir": "ltr",
+    "lang": "en-US"
+}
```
Comment (realslimshanky) [inline manifest.json]: Please add a newline here.

**Review by realslimshanky — CHANGES_REQUESTED** [review]

**Review by akshatnitd — COMMENTED** [review]
@harshit98 @vaibhavsingh97 @niteshkumarniranjan I can see that the files that are frequently changed are added in the `filesToCache` list. So, whenever we open the site, we need to first clear the cache to see the updated changes or change the `staticCacheName` on every deployment. Wanted your views, if some of the files should be removed from the list.

File: manifest.json [hunk 2]
```diff
@@ -0,0 +1,13 @@
+{
```
Comment (dilpreetsio) [inline manifest.json:1]: @niteshkumarniranjan Great work! Just one more thing, add icons to the manifest file.

Comment (niteshkumarniranjan) [reply manifest.json]: @akshatnitd No We don't have to clear the cache but instead we have to change the cache name.

File: sw.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+      'img/mentors/mariobehling.jpg',
+      'img/mentors/sumedh.jpeg',
+      'img/mentors/nikit.jpeg',
+      'img/mentors/hpdang.jpg',
+      'img/mentors/mohitsharma.png',
+      'img/slider/community-map.jpg',
+      'img/slider/group-photo.jpg'
+    ];
+    
+    var staticCacheName = 'pages-cache-v1';
```
Comment (niranjannitesh) [inline sw.js:45]: We have to change the staticCacheName after changes.

**Comment by niranjannitesh** [comment]
@geekyd Added Icons!

**Review by harshit98 — APPROVED** [review]
LGTM 👍

**Review by realslimshanky — APPROVED** [review]

**Pull request merged**